support/util/networking.go

```diff
@@ -121,3 +130,56 @@ func AllowedCIDRBlocks(hcp *hyperv1.HostedControlPlane) []hyperv1.CIDRBlock {
 	}
 	return nil
 }
+
+func SetAdvertiseAddresses(hcp *hyperv1.HostedControlPlane, defaults *DefaultAdvIps) []string {
+	addresses := make([]string, 0)
+
+	// User defined AdvertiseAddress/AdvertiseAddresses
+	if hcpAdvAddresses := AdvertiseAddresses(hcp); hcpAdvAddresses != nil {
+		fmt.Println("AdvertiseAddresses found:", hcpAdvAddresses)
+		addresses = hcpAdvAddresses
+	}
+
+	if hcpAdvAddress := AdvertiseAddress(hcp); hcpAdvAddress != nil {
+		fmt.Println("AdvertiseAddress found:", hcpAdvAddress)
+		addresses = append(addresses, *hcpAdvAddress)
+	}
+
+	// If there is not any defined AdvertiseAddress we should default them.
+	if len(addresses) == 0 {
+		fmt.Println("AdvertiseAddress or AdvertiseAddresses not found in hcp, using default values")
+		ipv4, err := IsIPv4(hcp.Spec.Networking.ServiceNetwork[0].CIDR.String())
+		if err != nil {
+			// if there is an error validating the ServiceNetworkCIDR we use the IPv4 default ip
+			fmt.Printf("error checking the ServiceNetworkCIDRs: %v", err)
+			return []string{defaults.IPv4}
```

Comment:
why is this ignoring error and returning default?

Reply:
This is the last mile, the `ServiceNetworkCIDRs` should be validated at this point, if not means that the api machinery didn't validate properly the networks and that should be happened before. This is only to default the value in case of any other validations failed during the HC deployment.